### DISCUSSION TITLE
Any gRPC related idea ?

### DISCUSSION THREAD

**Discussion by Clement-Jean** [SeaQL/summer-of-code#6] [Q&A]
gRPC web services are mentioned in your GSOC page but I do not see any idea related to it. Is there by any chance something that I could contribute to in that domain ?

I'm passionate about gRPC and protocol buffers, I even teach some Udemy course on the subject in different languages. Would be great if there was a way I can help.

Note: I'm aware that the gRPC implementation for Rust is not an official one.

**Comment by billy1624** [comment]
Hey @Clement-Jean, welcome! An extension that built on top of [SeaORM](https://github.com/SeaQL/sea-orm) would definitely be welcomed!

Talking about the gRPC implementation in Rust. Yes, there aren't any "official" one. However, I do found [Tonic](https://github.com/hyperium/tonic) and this issue on gRPC's GitHub
- https://github.com/grpc/grpc/issues/25937

**Reply by Clement-Jean** [reply]
Any mentor who could guide me through the project ? sounds like an interesting idea.

Yes there are two popular implementations. Tonic and https://github.com/stepancheg/grpc-rust

**Comment by tyt2y3** [comment]
I think there exists a very interesting projected based on tonic / gRPC called https://github.com/awto-rs/awto, although it may be inactive for now

We will still consider GSoC proposals in this topic, but 1) it has to have enough content to be qualified as a 350 hours project 2) it has to be original (and substantially different from awto) 3) at least have some community interest (please reply to this thread if you want to see it happen)

**Comment by Clement-Jean** [comment]
I think that in order to do something similar to what I saw from SeaORM, we could use protobuf options and reflection and from that we could generate some code from that. We could even generate code that is specific to SeaORM by writing a protoc plugin.  instead of just generating gRPC services or Messages, we could personalize the generated code

I only did that in C++ for now, never with rust but this might be interesting. I was doing something similar to [RoomDB](https://developer.android.com/training/data-storage/room)'s annotation but from protobuf.

**Reply by billy1624** [reply]
Hey @Clement-Jean, we got a gRPC example for SeaORM, see https://github.com/SeaQL/sea-orm/tree/master/examples/tonic_example

**Reply by Clement-Jean** [reply]
The example looks great, just one question, did you guys write the post.rs in the entity folder or was it generated ?

**Reply by billy1624** [reply]
We write it by hand but it can be generated, see https://www.sea-ql.org/SeaORM/docs/generate-entity/sea-orm-cli#generating-entity-files

**Comment by billy1624** [comment]
Hey @Clement-Jean, GSoC contributor application now opens. Please submit your proposal early so we can review and provide feedbacks :)
